### PR TITLE
Add webpack example

### PR DESCRIPTION
**Pull request by VictorBjelkholm**
Comes from https://github.com/VictorBjelkholm/webpack-node-ipfs-api-demo which I'll deprecate if we decide to merge this.

### CODE REVIEW

File: examples/webpack/.jshintrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "node": true,
+  "browser": true,
+  "esnext": true,
+  "newcap": false
+}
```
Comment (dignifiedquire) [inline examples/webpack/.jshintrc]: do we really need jshint and eslint?

File: examples/webpack/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "react-hot-boilerplate",
```
Comment (dignifiedquire) [inline examples/webpack/package.json]: lets rename this properly

File: examples/webpack/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  "version": "1.0.0",
+  "description": "Boilerplate for ReactJS project with hot code reloading",
```
Comment (dignifiedquire) [inline examples/webpack/package.json]: can update this as well

File: examples/webpack/package.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    "reactjs",
+    "boilerplate",
+    "hot",
+    "reload",
+    "hmr",
+    "live",
+    "edit",
+    "webpack"
+  ],
+  "author": "Dan Abramov <dan.abramov@me.com> (http://github.com/gaearon)",
```
Comment (dignifiedquire) [inline examples/webpack/package.json]: I don't think Dan wrote this ;)

File: examples/webpack/src/App.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react'
+
+import ipfsAPI from 'ipfs-api'
+
+var ipfs = ipfsAPI('localhost', '5001')
+const stringToUse = 'Hello World From Webpack'
```
Comment (dignifiedquire) [inline examples/webpack/src/App.js]: lets make it `hello world from wepbacked IPFS`

File: examples/webpack/src/App.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        res.on('data', (d) => {
+          data = data + d
+        })
+        res.on('end', () => {
+          this.setState({added_file_contents: data})
+        })
+      })
+      ipfs.dht.findprovs(hash, (err, res) => {
+        if (err) throw err
+        console.log('found provs!')
```
Comment (dignifiedquire) [inline examples/webpack/src/App.js]: why not write this to the dom like the others do?

**Comment by VictorBjelkholm** [comment]
@dignifiedquire thanks for the review. The example comes from a quick setup I did, coming from Dan's react hot boilerplate. I'll address the things you've mentioned, and also remove the findprovs part (since it's out of scope of the example). Thanks!

**Comment by diasdavid** [comment]
Sweeeet! :D 

Can we call the folder `bundle-webpack` so that then we also have `bundle-browserify`? 

Also, can you add a README just explaining what people are doing with all of this pieces? Or even just say? "Copy this to your workspace and you get it working by running X Y Z commands"

**Comment by VictorBjelkholm** [comment]
@diasdavid you got it! :+1:

**Comment by diasdavid** [comment]
awesome. Also update the readme section https://github.com/ipfs/js-ipfs-api#in-a-web-browser-through-browserify to point to both examples :)

**Comment by VictorBjelkholm** [comment]
Thanks for the feedback, everything have been addressed now!

@diasdavid @dignifiedquire please take a look again and merge if :+1:

**Comment by VictorBjelkholm** [comment]
@diasdavid linting should pass now, yes.

**Pull request merged**